### PR TITLE
VxScan: Automatically Set Precinct if Only One

### LINKED ISSUE #2636: VxScan: do not require configuration if election definition only has one precinct

### PR DESCRIPTION
**Pull request by adghayes**
## Overview
Closes #2636. If an election definition has only one precinct, then that individual precinct is set as the VxScan's precinct after the ballot package is done loading. This will bypass the "No Precinct Selected" screen. 

In the case of having only one precinct, I've removed the "All Precincts" option from VxScan.

## Demo Video or Screenshot
Not very much to show.

## Testing Plan 
- Added test for augmented `addTemplates` method
- Added test for `ElectionManagerScreen` in case of only one precinct

## Checklist
- [ ] I have added [logging](https://github.com/votingworks/vxsuite/tree/main/libs/logging) where appropriate to any new user actions, system updates such as file reads or storage writes, or errors introduced.
- [ ] I have added JSDoc comments to any newly introduced exports

- [ ] I have added a screenshot and/or video to this PR to demo the change
- [x] I have added the "user_facing_change" label to this PR to automate an announcement in #machine-product-updates

### CODE REVIEW

**Comment by adghayes** [comment]
Reviewers: review the two commits as one change. The first one is a relic of a different approach which I decided against (but has more integration testing).

File: frontends/precinct-scanner/src/screens/election_manager_screen.tsx [hunk 1]
```diff
@@ -156,10 +158,16 @@ export function ElectionManagerScreen({
             onChange={changeAppPrecinctSelection}
             large
           >
-            <option value="" disabled>
-              Select a precinct for this device…
-            </option>
-            <option value={ALL_PRECINCTS_OPTION_VALUE}>All Precincts</option>
+            {!precinctSelectionValue && (
```
Comment (jonahkagan) [inline frontends/precinct-scanner/src/screens/election_manager_screen.tsx:161]: What does this change do?

File: frontends/precinct-scanner/src/api/config.ts [hunk 2]
```diff
@@ -249,6 +254,12 @@ export function addTemplates(pkg: BallotPackage): AddTemplatesEvents {
         });
       }
 
+      // Set the precinct if there is only one precinct
+      if (pkg.electionDefinition.election.precincts.length === 1) {
+        const onlyPrecinct = pkg.electionDefinition.election.precincts[0];
+        await setPrecinctSelection(singlePrecinctSelectionFor(onlyPrecinct.id));
+      }
```
Comment (jonahkagan) [inline frontends/precinct-scanner/src/api/config.ts:261]: How did you choose to add this logic to `addTemplates`? It seems a bit unrelated to the templates logic.

A few alternative ideas:
- Rename this function to better represent what it's doing holistically
- Move this logic to somewhere else (maybe where `addTemplates` is called)
- Move this logic to the backend (since that's the general direction we're moving in)

File: frontends/precinct-scanner/src/screens/election_manager_screen.tsx (hunk 1 above)
Comment (adghayes) [reply frontends/precinct-scanner/src/screens/election_manager_screen.tsx:161]: Hides the `Select a precinct for this device...` text when the precinct is already set, because it's just a disabled option in the list. A bit unrelated, and happy to reverse.

File: frontends/precinct-scanner/src/api/config.ts (hunk 2 above)
Comment (adghayes) [reply frontends/precinct-scanner/src/api/config.ts:261]: I chose to add it to to `addTemplates` because it's also where the election definition is set, so it seemed appropriate to set the precinct here as well. Originally (in the first commit) I had it where `addTemplates` is called. I don't think it was much clearer, and it was way harder to test. `addTemplates` should probably be renamed.

I prefer the idea of moving it to the backend, I'll look at that.

File: frontends/precinct-scanner/src/screens/election_manager_screen.tsx (hunk 1 above)
Comment (jonahkagan) [reply frontends/precinct-scanner/src/screens/election_manager_screen.tsx:161]: Cool, that seems fine

**Review by jonahkagan — APPROVED** [review]
